src/io/syncsocket.c

```diff
@@ -306,41 +293,57 @@ struct sockaddr * MVM_io_resolve_host_name(MVMThreadContext *tc, MVMString *host
         case SOCKET_FAMILY_INET6:
             hints.ai_family = AF_INET6;
             break;
-        case SOCKET_FAMILY_UNIX:
-            hints.ai_family = AF_UNIX;
-            break;
+        case SOCKET_FAMILY_UNIX: {
+            /* TODO: UNIX socket support exists in newer versions of Windows.
+             *       See if it's good enough for us to use. */
+#ifndef AF_UNIX
+            MVM_free(host_cstr);
+            MVM_exception_throw_adhoc(tc, "UNIX sockets are not supported by MoarVM on this platform");
+#else
+            static const size_t MAX_SUN_LEN = sizeof(((struct sockaddr_un *)NULL)->sun_path);
+            size_t              sun_len     = strnlen(host_cstr, MAX_SUN_LEN);
+
+            if (sun_len >= MAX_SUN_LEN) {
+                char *waste[] = { host_cstr, NULL };
+                MVM_exception_throw_adhoc_free(
+                    tc, waste, "Socket path '%s' is too long (max supported by this platform is %zu)",
+                    host_cstr, MAX_SUN_LEN
+                );
+            } else {
+                struct sockaddr_un *result_un = MVM_malloc(sizeof(struct sockaddr_un));
+                result_un->sun_family         = AF_UNIX;
+                strcpy(result_un->sun_path, host_cstr);
+                MVM_free(host_cstr);
+                return (struct sockaddr *)result_un;
+            }
+#endif
+        }
         default:
-            MVM_exception_throw_adhoc(tc, "Unsupported socket family: %hu", family);
+            MVM_exception_throw_adhoc(tc, "Unsupported socket family: %"PRIu16"", family);
             break;
     }
 
-    hints.ai_socktype = 0;
-    hints.ai_flags = AI_ADDRCONFIG | AI_NUMERICSERV | AI_PASSIVE;
-    hints.ai_protocol = 0;
-    hints.ai_addrlen = 0;
-    hints.ai_addr = NULL;
-    hints.ai_canonname = NULL;
-    hints.ai_next = NULL;
-
     snprintf(port_cstr, 8, "%d", (int)port);
 
     MVM_gc_mark_thread_blocked(tc);
-    error = getaddrinfo(host_cstr, port_cstr, &hints, &result);
+    e = getaddrinfo(host_cstr, port_cstr, &hints, &result);
```

Comment:
Why the rename of the variable? Seems to make the code less readable for no good reason.

Reply:
I had the impression naming it `e` is better style when writing C, but checking how the rest of MoarVM names errors little else names it like this.